compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -4980,6 +4981,29 @@ declare_lint! {
     "detects pointer to integer transmutes in const functions and associated constants",
 }
 
+declare_lint! {
+    /// The `redundant_transmutation` lint detects transmutations that have safer alternatives.
+    ///
+    /// ### Example
+    ///
+    /// ```rust
+    /// fn bytes_at_home(x: [u8; 4]) -> u32 {
+    ///   unsafe { std::mem::transmute(x) }
+    /// }
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// People dont realize that safer methods such as
+    /// [`u32::to_ne_bytes`](https://doc.rust-lang.org/std/primitive.u32.html#method.to_ne_bytes) exist,
+    /// so this lint exists to lint on cases where people write transmutes that dont need to be there.
```

Comment:
```suggestion
    /// Using safe methods is preferable over calls to `transmute` as
    /// they more clearly communicate the intent, are easier to review, and
    /// are less likely to accidentally result in unsoundness.
```
I feel like you have a subtly wrong perspective here :thinking: 
- your explanation justifies the lint to the project, as in, why does this lint exist
- as this is user-facing, we should instead explain why this lint matters for the user, as in, why the suggested alternative is preferable

Not sure if that distinction makes sense to you :thinking:

Reply:
Yeah that makes sense to me; i just sort of wrote it from the issue